Cython/Utility/CommonStructures.c

```diff
@@ -16,16 +16,31 @@ static PyTypeObject* __Pyx_FetchCommonTypeFromSpec(PyObject *module, PyType_Spec
 //@requires:ExtensionTypes.c::FixUpExtensionType
 //@requires: FetchSharedCythonModule
 //@requires:StringTools.c::IncludeStringH
+//@requires:Optimize.c::dict_setdefault
 
 static int __Pyx_VerifyCachedType(PyObject *cached_type,
                                const char *name,
-                               Py_ssize_t basicsize,
                                Py_ssize_t expected_basicsize) {
+    Py_ssize_t basicsize;
+
     if (!PyType_Check(cached_type)) {
         PyErr_Format(PyExc_TypeError,
             "Shared Cython type %.200s is not a type object", name);
         return -1;
     }
+
+#if CYTHON_COMPILING_IN_LIMITED_API
+    PyObject *py_basicsize;
+    py_basicsize = PyObject_GetAttrString(cached_type, "__basicsize__");
+    if (unlikely(!py_basicsize)) return -1;
+    basicsize = PyLong_AsSsize_t(py_basicsize);
+    Py_DECREF(py_basicsize);
+    py_basicsize = 0;
```

Comment:
I think we always use `NULL` in hand written code.
```suggestion
    py_basicsize = NULL;
```